packages/netlify-cms-lib-auth/src/pkce-oauth.js

```diff
@@ -3,33 +3,49 @@ import trimEnd from 'lodash/trimEnd';
 import { createNonce, validateNonce, isInsecureProtocol } from './utils';
 
 async function sha256(text) {
-  return Array.from(
-    new Uint8Array(await window.crypto.subtle.digest('SHA-256', new TextEncoder().encode(text))),
-  )
-    .map(el => {
-      return String.fromCharCode(el);
-    })
-    .join('');
+  const encoder = new TextEncoder();
+  const data = encoder.encode(text);
+  const digest = await window.crypto.subtle.digest('SHA-256', data);
+  const sha = String.fromCharCode(...new Uint8Array(digest));
+  return sha;
 }
 
-//based on https://github.com/auth0/auth0-spa-js/blob/9a83f698127eae7da72691b0d4b1b847567687e3/src/utils.ts#L147
+// based on https://github.com/auth0/auth0-spa-js/blob/9a83f698127eae7da72691b0d4b1b847567687e3/src/utils.ts#L147
 function generateVerifierCode() {
-  const randomValuesArr = new Uint8Array(128);
-  return Array.from(window.crypto.getRandomValues(randomValuesArr))
+  // characters that can be used for codeVerifer
+  // excludes _~ as if included would cause an uneven distribution as char.length would no longer be a factor of 256
+  const chars = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-.';
+  const randomValues = window.crypto.getRandomValues(new Uint8Array(128));
+  return randomValues
     .map(val => {
-      // characters that can be used for codeVerifer
-      // excludes _~ as if included would cause an uneven  distribution as char.length would no longer be a factor of 256
-      const chars = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-.';
       return chars[val % chars.length];
     })
     .join('');
 }
 
-function createOrGetCodeVerifier() {
```

Comment:
I separated this function to make it more explicit.
Before it seems that creating a code verifier in `completeAuth` could be consider as a valid use case (am I assuming correctly that it is not?)